Bump TS lib to ES2022

Pl217 opened the pull request:
Our code already used ES2022 feature with `String.prototype.at()`, but the fact that we're not using the right library was masked by package `@types/node` having a polyfill for this method, which [was removed](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72589/commits/1380e73168a48eb3b7933a969bce8cc949ef4560) in v24 of that package.

Also, I've modified CI to now actually try and build both apps, so that we don't fail to catch problems like this during review phase in the future.

Pl217 committed:
Bump TS lib to ES2022

Our code already used ES2022 feature with
`String.prototype.at()`, but the fact that
we're not using the right library was masked
by package `@types/node` having a polyfill
for this method, which was removed in v24
of that package
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72589/commits/1380e73168a48eb3b7933a969bce8cc949ef4560

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review

Pl217 committed:
Revert "Fetch all commits needed for `nrwl/nx-set-shas`"

This reverts commit f7ec554dfa088650f0c29594647b6510fb6058fb.

Pl217 committed:
Always build all apps in CI

We've seen that trying to build only affected
apps is not helpful, because it usually matches
no app. Since we've stopped using `nrwl/nx-set-shas`
action to help set head and base SHAs for
determining what's affected, we now just
build all (both) apps (in parallel)

Action `nrwl/nx-set-shas` was added in:
https://github.com/UN-OCHA/hpc-cdm/pull/445
removed in:
https://github.com/UN-OCHA/hpc-cdm/pull/493

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review

Onitoxan commented on .github/workflows/workflow.yml:
```diff
@@ -38,7 +38,7 @@ jobs:
         run: npm run lint
 
       - name: Build affected apps
-        run: npx nx affected:build
```
Nitpick: We should change the name from `Build affected apps` to `Build apps` or whatever you think suits best.

Beside this, should we modify also `npm run lint` to lint all the apps? Right now just does on affected. I was thinking that if we update `hpc-repo-tools` there are some new eslint rules being added so it would be nice that it always runs on every project to catch these cases. What do you think?

Pl217 replied:
Good nitpick about the step name :)

Regarding linting, it runs `lint` script, which is defined as `"lint": "nx affected:lint && nx lint && prettier --check '**/*' '*'"`, which runs linting on both affected and all apps. I don't know why both, but I will not touch it.

The library `hpc-repo-tools` is already updated to its latest version `8.0.0`

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review
